2019/es/src/0xa2-broken-user-authentication.md

```diff
@@ -0,0 +1,76 @@
+
+## ¿Es vulnerable la API?
+
+Los recursos y flujos de autenticación son activos que deben ser protegidos. "Olvidé mi contraseña / recuperar contraseña" debe ser tratado de la misma forma que los mecanismos de autenticación.
+
+Una API es vulnerable si:
+* Permite el [relleno de credenciales][1] donde el atacante tiene una lista de usuarios y contraseñas válidos.
+* Permite a los atacantes realizar ataques de fuerza bruta sobre la misma cuenta de usuario, sin presentar mecanismos de bloqueo de captcha/cuenta.
+* Permite contraseñas débiles.
+* Envía detalles de autenticación sensibles, como tokens de authenticación y contraseñas en la URL.
+* No valida la autenticidad de los tokens.
+* Acepta tokens JWT no/débilmente firmados (`"alg":"none"`)/no valida su fecha de expiración.
+* Usa contraseñas en texto plano, encriptadas o codificadas con algoritmos de hash débiles.
+* Usa claves de encriptación débiles.
+
+## Escenarios de Ataque de Ejemplo
+
+## Escenario #1
+
+[Relleno de credenciales][1] (usando [listas de usuarios/contraseñas conocidas][2]), es un ataque común. 
+Si una aplicación no implementa protecciones de ataque automático o de relleno de credenciales, 
+la aplicación puede ser usada como un oráculo de contraseñas (probador) para determinar si las credenciales son válidas.
+
+## Escenario #2
+
+Un atacante comienza el flujo de recuperación de contraseña realizando una petición POST a `/api/system/verification-codes`
+e incluyendo el nombre de usuario en el cuerpo de la petición. 
+A continuación un token SMS con 6 dígitos es enviado al teléfono de la víctima. 
+Como la API no implementa una política de límite de peticiones, el atacante puede probar todas las combinaciones posibles usando 
+un script multi-hilo contra el recurso `/api/system/verification-codes/{smsToken}` para descubrir el token correcto en unos minutos.
+
+## Cómo prevenir
+
+* Asegurarse que conoce todos los flujos de autenticación posibles a la API (móviles/
+  web/links profundos que implementan autenticación en 1 click/etc.)
+* Pregúntele a sus ingenieros que flujos ha olvidado.
```

Comment:
"qué", not "que"

Reply:
Should be "Asegurarse qué flujos de autenticación..." to have accent. 

I think this one is correct as it is now.

Reply:
I refer to "Pregúntele a sus ingenierós QUÉ flujos ha olvidado"

Reply:
Missed that one, thanks!